### PR TITLE
Implement du threshold argument handling

### LINKED ISSUE #2352: `du --threshold=SIZE` isn't implemented
```
       -t, --threshold=SIZE
              exclude  entries  smaller  than  SIZE  if  positive,  or entries
              greater than SIZE if negative
```
it should have also `--th` as alias

Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/du/src/du.rs
tests in:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_du.rs

### PR DESCRIPTION
**Pull request by siebenHeaven**
This PR implements --threshold argument handling for du utility and closes #2352.

Marking as draft as tests are yet to be added.

### CODE REVIEW

File: src/uu/du/src/du.rs
Comment (siebenHeaven) [inline src/uu/du/src/du.rs:266]: TODO: Move after get_block_size_from_env (for grouping block size related functions together)

File: src/uu/du/src/du.rs
Comment (siebenHeaven) [inline src/uu/du/src/du.rs:717]: TODO: Switch to positive logic for readability?

**Comment by sylvestre** [comment]
Some tasks are failing but I guess you know about this ;)

**Comment by jhscheer** [comment]
Hi, please wait until #2297 gets merged.

It unifies handling of the `SIZE` parameter and should make implementing '--threshold' easier.

e.g. `read_block_size` will change to this:
```rust
use uucore::parse_size::{parse_size, ParseSizeError};
fn read_block_size(s: Option<&str>) -> usize {
    if let Some(s) = s {
        parse_size(s).map_or_else(
            |e| crash!(1, "{}", format_error_message(e, s, options::BLOCK_SIZE)),
            |n| n,
        )
    }
[...]

fn format_error_message(error: ParseSizeError, s: &str, option: &str) -> String {
    match error {
        ParseSizeError::ParseFailure(_) => format!("invalid --{} argument '{}'", option, s),
        ParseSizeError::SizeTooBig(_) => format!("--{} argument '{}' too large", option, s),
    }
}
```

**Comment by siebenHeaven** [comment]
> Some tasks are failing but I guess you know about this ;)

Actually, there was a logical bug which I'd missed. I want to add a test for this specific functionality before asking for this to be merged which is what I meant before.



> Hi, please wait until #2297 gets merged.

Thanks, I will wait for that to get merged and then rebase on top of it + add a test for this.

**Comment by siebenHeaven** [comment]
Rebased on top of latest parse_size changes and added couple of tests for the feature - this PR is now ready!

File: src/uu/du/src/du.rs
Comment (tertsdiepraam) [inline src/uu/du/src/du.rs:746]: I had to think for a minute about what the `bool` meant. Maybe it makes sense to make this an `enum` like this to provide a bit more context?
```rust
enum Threshold {
    Less(u64),
    Greater(u64),
}
```

File: src/uu/du/src/du.rs
Comment (tertsdiepraam) [inline src/uu/du/src/du.rs:752]: You could combine the `starts_with` calls into `s.starts_with(&['-', '+'])`, I think. Additionally, you could cast it to a `usize`, but that is arguably less readable.

File: src/uu/du/src/du.rs
Comment (tertsdiepraam) [inline src/uu/du/src/du.rs:767]: I would prefer words as variable names, so `size` in this case.

**Review by tertsdiepraam — COMMENTED** [review]
Nice! And great use of `parse_size`! I left a few minor suggestions, but they're more stylistic choices I guess, so you can ignore them if you don't agree :)

File: src/uu/du/src/du.rs
Comment (siebenHeaven) [reply src/uu/du/src/du.rs:746]: Good suggestion - I'll give this a shot.

File: src/uu/du/src/du.rs
Comment (siebenHeaven) [reply src/uu/du/src/du.rs:752]: Ah - will give it a shot.

File: src/uu/du/src/du.rs
Comment (siebenHeaven) [reply src/uu/du/src/du.rs:767]: Will do.

File: src/uu/du/src/du.rs
Comment (jhscheer) [reply src/uu/du/src/du.rs:752]: FYI:
there was a similar heuristic required in head/tail (and maybe others, idr):
https://github.com/uutils/coreutils/blob/440eba628c6c77875013afc4ade1b32f3f77c017/src/uu/tail/src/tail.rs#L406-L423

Comment (siebenHeaven) [reply src/uu/du/src/du.rs]: Wondering if we could write another common wrapper (parse_signed_size ?) :)

Comment (jhscheer) [reply src/uu/du/src/du.rs]: I was also considering this while writing `parse_size`, but decided against it since (at that time) I only knew about 2 files (with ~5 duplicate LOCs) that would've been applicable for this.
If you like to keep this consistent across uutils, a common wrapper would be nice. However, just copying the ~5 LOCs would also be fine.

**Review by tertsdiepraam — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**
